client/extensions/zoninator/components/forms/zone-details-form/index.jsx

```diff
@@ -0,0 +1,71 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { reduxForm } from 'redux-form';
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import CompactCard from 'components/card';
+import FormButton from 'components/forms/form-button';
+import FormTextInput from 'components/forms/form-text-input';
+import FormTextarea from 'components/forms/form-textarea';
+import ReduxFormFieldset from 'components/redux-forms/redux-form-fieldset';
+import SectionHeader from 'components/section-header';
+
+const form = 'extensions.zoninator.zoneDetails';
+
+const ZoneDetailsForm = ( {
+	handleSubmit,
+	label,
+	onSubmit,
+	submitting,
+} ) => {
+	const save = data => onSubmit( form, data );
+
+	return (
+		<form onSubmit={ handleSubmit( save ) }>
+			<SectionHeader label={ label }>
+				<FormButton
+					compact
+					disabled={ submitting }
+					isSubmitting={ submitting }>
+					{ translate( 'Save' ) }
+				</FormButton>
+			</SectionHeader>
+			<CompactCard>
+				<ReduxFormFieldset
+					name="name"
+					label={ translate( 'Zone name' ) }
+					component={ FormTextInput } />
+				<ReduxFormFieldset
+					name="description"
+					label={ translate( 'Zone description' ) }
+					component={ FormTextarea } />
+			</CompactCard>
+		</form>
+	);
+};
+
+ZoneDetailsForm.propTypes = {
+	label: PropTypes.string.isRequired,
```

Comment:
Should probably also include `handleSubmit` and `submitting` since those are being used as well.